pkg/providers/openid/openid.go

```diff
@@ -57,6 +57,29 @@ func (Provider) GetUserInfo(r *http.Request, user *structs.User, customClaims *s
 		log.Error(err)
 		return err
 	}
+	var f interface{}
+	err = json.Unmarshal(data, &f)
+	if err != nil {
+		log.Error("Error unmarshaling claims")
+		return err
+	}
+	if cfg.Cfg.TeamWhiteListClaim != "" {
+		log.Infof("TeamWhiteListClaim is %+v", cfg.Cfg.TeamWhiteListClaim)
+		m := f.(map[string]interface{})
+		for k := range m {
+			log.Infof("checking claim %s", k)
+			if k == cfg.Cfg.TeamWhiteListClaim {
+				/*for _, membership := range m[k].([]interface{}) {
+					user.TeamMemberships = append(user.TeamMemberships, membership.(string))
+				}*/
+				log.Infof("claim values of %+v is %+v, converted to %+v", k, m[k], m[k].(string))
+				user.TeamMemberships = append(user.TeamMemberships, m[k].(string))
```

Comment:
is permission name always sent as string ? should check if type is string before casting and appending ?